### DISCUSSION TITLE
field  Date::make

### DISCUSSION THREAD

**Discussion by Mr-Coquin** [moonshine-software/moonshine#1058] [Ideas]
Добавить возможность ввода  разных вариантов формата дат: года и год + месяц (например год: 2024,2025 и т.д. , месяц:03.2024 (март 2024), 09.2025 (сентябрь 2025)).

**Comment by shevl-dev** [comment]
Ну сейчас же там параметром можно передавать format? Или ты предлагаешь добавить сахарные fluent методы для установки разных стандартных форматов?

**Reply by Mr-Coquin** [reply]
> Ну сейчас же там параметром можно передавать format? Или ты предлагаешь добавить сахарные fluent методы для установки разных стандартных форматов?

У меня есть задача к примеру в некоторых полях выбирать только год пользователю в форме . Ему месяц и день не нужен. Сейчас я обыгрываю это так к примеру из одной формы у меня на проекте(тут только текущий год можно взять или следующий), но в будущем может понадобиться к примеру выбрать только 2015(да можно через rules с маской - но это не эстетично):
                Number::make('year')
                    ->min((int)date("Y"))
                    ->max((int)date("Y") + 1)
                    ->step(1)
                    ->default((int)$this->getResource()->year)
                    ->buttons()

**Reply by DissNik** [reply]
Date использует `<input type="date">`
input типа year в html нет, есть только week и month, которые можно назначить например полю Text через кастомные атрибуты

Так же можно сделать кастомное поле которое будет использовать какой-нибудь datepicker на js

**Reply by Mr-Coquin** [reply]
> Date использует `<input type="date">` input типа year в html нет, есть только week и month, которые можно назначить например полю Text через кастомные атрибуты


Очень сомневаюсь, как это будет работать . если 2 type будут в input только если там проверка кастом идет на второй input type 

Но это сообщение поэтому и находиться в Идеях

**Reply by DissNik** [reply]
```
Text::make('Date')
        ->customAttributes([
            'type' => 'month'
        ])
```
Будет работать

А в коробке такое поле маловерятно что появится, только как пакет
